packages/flutter/test/material/data_table_test.dart

```diff
@@ -157,6 +157,21 @@ void main() {
 
     expect(log, <String>['select-all: false']);
     log.clear();
+
+    // Tapping the parent checkbox when all rows are selected and one is
+    // disabled, deselects all.
+    await tester.pumpWidget(MaterialApp(
+      home: Material(
+        child: buildTable(
+          <bool>[true, true, false],
+          disabledIndex: 2,
```

Comment:
does it matter that the disabled index is already false? make it true to be certain that it goes back to false?

Reply:
If I put it to `true` here, the test would pass without my change as all checkboxes would be selected, and the parent therefore would deselect all (without ignoring any disabled checkboxes)

By putting it to `false` here, I am making sure my change works.